internal/services/network/application_gateway_resource.go

```diff
@@ -4851,7 +5551,7 @@ func applicationGatewayProbeHash(v interface{}) int {
 			buf.WriteString(fmt.Sprintf("%d", v.(int)))
 		}
 		if match, ok := m["match"]; ok {
-			if attrs := match.([]interface{}); len(attrs) == 1 {
+			if attrs := match.([]interface{}); len(attrs) == 1 && attrs[0] != nil {
```

Comment:
Is there a specific reason to add this? As there is `required` property in the `match` block, all its elements should be non nil.